tool/src/cpp/formatter.rs

```diff
@@ -51,32 +67,70 @@ impl<'tcx> Cpp2Formatter<'tcx> {
     }
 
     /// Resolve and format the name of a type for use in header names
-    pub fn fmt_decl_header_path(&self, id: TypeId) -> String {
-        let resolved = self.c.tcx().resolve_type(id);
-        let type_name = resolved
-            .attrs()
-            .rename
-            .apply(resolved.name().as_str().into());
-        if let Some(ref ns) = resolved.attrs().namespace {
+    pub fn fmt_decl_header_path(&self, id: SymbolId) -> String {
+        let (name, namespace) = match id {
+            SymbolId::FunctionId(f) => {
+                let resolved = self.c.tcx().resolve_function(f);
+                let namespaced = if let Some(ns) = &resolved.attrs.namespace {
+                    format!("_{}", ns.replace("::", "_"))
+                } else {
+                    "".into()
+                };
+                (
+                    format!("free{namespaced}_functions"),
```

Comment:
nit: the folder already sets the namespace, we don't need to additionally pass in `{namespaced}`